src/test/fuzz/simplicity.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <primitives/transaction.h>
+extern "C" {
+#include <simplicity/elements/env.h>
+#include <simplicity/elements/exec.h>
+}
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+FUZZ_TARGET(simplicity)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    {
+        const std::optional<CMutableTransaction> mtx_precomputed = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+        if (mtx_precomputed) {
+            const CTransaction tx_precomputed{*mtx_precomputed};
+            const PrecomputedTransactionData precomputed_transaction_data{tx_precomputed};
+            const transaction* tx = precomputed_transaction_data.m_simplicity_tx_data;
```

Comment:
I've finally got the coverage analysis of fuzzing working, and I will share how to do that with you and everyone else shortly.

However, in the meantime, the results I have suggest to me that `tx` is always `NULL`, and thus the fuzzer doesn't even make it past the first line of `elements_simplicity_execSimplicity`

Reply:
oh interesting. maybe if we provide a well formed transaction to the fuzzer via the corpus it will have a better starting point

Reply:
You can probably go ahead and check the return value of  `elements_simplicity_execSimplicity` and assert that it is true.  `elements_simplicity_execSimplicity` only returns `false` during catastrophic failure, such as `malloc` failing, or, as in this case, if you have called it with (detectable) incorrect parameters.

The implementation actually does this assertion: https://github.com/ElementsProject/elements/blob/2423e8559ab4c25287c1ecd5add02119f7cc11c6/src/script/interpreter.cpp#L3115-L3117